### PR TITLE
Dependency and README updates

### PR DESCRIPTION
**Pull request by nblumhardt**
This PR adds `Console`, `Debug`, and `File` sinks to the package dependencies, along with _Serilog.Formatting.Compact_ (JSON) and _Serilog.Settings.Configuration_. The README is updated accordingly.

The goal is to make the onboarding process smoother. There's still work to do, including sample updates to match the ASP.NET Core 3.0 hosting startup.

Partial #95

### CODE REVIEW

File: samples/EarlyInitializationSample/Program.cs [hunk 1]
```diff
@@ -21,6 +21,7 @@ public static int Main(string[] args)
             Log.Logger = new LoggerConfiguration()
                 .ReadFrom.Configuration(Configuration)
                 .Enrich.FromLogContext()
+                .WriteTo.Debug()
```
Comment (merbla) [inline samples/EarlyInitializationSample/Program.cs]: I like this, there are lots of questions when folks are getting started that this may help.

Comment (nblumhardt) [reply samples/EarlyInitializationSample/Program.cs]: :+1:

**Comment by nblumhardt** [comment]
The reason I left this in draft is that it will update the README in a way that's incompatible with the current released package. We could just jump ahead, however, and release our 3.0 with the framework dependencies left at 2.2 (it doesn't look like we'll bump dependency versions this time around). Thoughts?

**Comment by SimonCropp** [comment]
> We could just jump ahead

:shipit:

**Comment by nblumhardt** [comment]
Will just need: https://github.com/serilog/serilog-extensions-hosting/pull/11

**Comment by nblumhardt** [comment]
Planning to merge and release this via 3.0.0, and only merge the example updates in #111 once .NET Core 3 has shipped.

**Pull request merged**